src/Libraries/Microsoft.Extensions.AI.Abstractions/Contents/DataContent.cs

```diff
@@ -183,6 +183,13 @@ public string Uri
     [JsonIgnore]
     public string MediaType { get; }
 
+    /// <summary>Gets or sets an optional file name associated with the data.</summary>
+    /// <remarks>
+    /// A service might use this name as part of citations or to help infer the type of data
+    /// being represented based on the file extension.
+    /// </remarks>
+    public string? FileName { get; set; }
```

Comment:
"FileName" sounds somewhat specific to local filesystems. Should we be calling it "Uri" instead?

Reply:
I take it though that you're intending this contain just the name, i.e. `payments.csv` as opposed to a full path that might reveal more context i.e. `/contoso/payroll/payments.csv`?

Reply:
> I take it though that you're intending this contain just the name, i.e. payments.csv

Yup.

We could just call it Name if that's desirable... I went back and forth on that.

Suggestions?

Reply:
Name sounds good